lib/truffle/truffle/cext.rb

```diff
@@ -1801,20 +1770,14 @@ def rb_thread_call_without_gvl(function, data1, unblock, data2)
     Thread.current.unblock unblocker, runner
   end
 
-  def rb_iterate_call_block( iter_block, block_arg, arg2, &block)
-    execute_with_mutex iter_block, block_arg, arg2, &block
-  end
-
-  def rb_iterate(function, arg1, iter_block, arg2, block)
-    if block
-      call_c_with_block function, arg1 do |block_arg|
-        rb_iterate_call_block(iter_block, block_arg, arg2) do |*args|
-          block.call(*args)
-        end
+  def rb_iterate(iteration, iterated_object, callback, callback_arg)
+    if (block = rb_block_proc)
```

Comment:
This looks strange to me. At the very least, the surrounding parens shouldn't be necessary. I'd maybe add parens to the function call to call out the fact it is is a function call. It's not clear to me at first glance that this is a predicate with side effects.

Reply:
This style is never used so far, and I believe we agreed it looks bad and doesn't carry well the intention (parenthesis are normally used for grouping expressions in conditions).
Can you make the assignment another line or drop the `()`?

Reply:
Yeah I don't like the look of it either.

Reply:
ok fixed